src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -1440,21 +1452,24 @@ export function registerTerminalActions() {
 				}
 			});
 		}
-		async run(accessor: ServicesAccessor, profile?: ITerminalProfile) {
-			const terminalService = accessor.get(ITerminalService);
+		async run(accessor: ServicesAccessor, optionsOrProfile?: ICreateTerminalOptions | ITerminalProfile) {
 			const commandService = accessor.get(ICommandService);
-			await terminalService.doWithActiveInstance(async t => {
-				const cwd = await getCwdForSplit(terminalService.configHelper, t, accessor.get(IWorkspaceContextService).getWorkspace().folders, accessor.get(ICommandService));
-				if (cwd === undefined) {
-					return undefined;
-				}
-				if (t.target === TerminalLocation.Editor) {
-					commandService.executeCommand('workbench.action.splitEditor');
-				} else {
-					terminalService.splitInstance(t, profile, cwd);
-					return accessor.get(ITerminalGroupService).showPanel(true);
-				}
-			});
+			const terminalGroupService = accessor.get(ITerminalGroupService);
+			const terminalService = accessor.get(ITerminalService);
+			const workspaceContextService = accessor.get(IWorkspaceContextService);
+			const options = convertOptionsOrProfileToOptions(optionsOrProfile);
+			const activeInstance = terminalService.getInstanceHost(options?.target).activeInstance;
+			if (!activeInstance) {
+				return;
+			}
+			const cwd = await getCwdForSplit(terminalService.configHelper, activeInstance, workspaceContextService.getWorkspace().folders, commandService);
+			if (cwd === undefined) {
+				return undefined;
+			}
+			const instance = terminalService.splitInstance(activeInstance, options?.config, cwd);
```

Comment:
I think we want to call the editor split command if the target is the editor?

Reply:
This is all built into TerminalService now, so no more accidents 🙂

Reply:
never mind I see you moved it!